app/code/Magento/InventoryCatalog/Exception/InvalidSourceAssignmentException.php

```diff
@@ -0,0 +1,14 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Exception;
+
+use Magento\Framework\Exception\InputException;
+
+class InvalidSourceAssignmentException extends InputException
```

Comment:
Looks like we don't need to introduce new exception type because we don't have processing of this exception in our code.

For calls in client code, this exception will be wrapper in \Exception in `\Magento\Inventory\Model\StockSourceLink\Command\AssignSourcesToStock::execute`